sidechain/migrations/2_deploy_dependency_contracts.js

```diff
@@ -0,0 +1,39 @@
+let Group = artifacts.require('Group');
+let Storage = artifacts.require('Storage');
+let OnlyOwnerAdminController = artifacts.require('OnlyOwnerAdminController');
+
+async function performMigration(deployer, network, accounts) {
+    await deployer.deploy(Storage);
```

Comment:
@wilbarnes 
Just realised currently this file is deploying all the contracts.
In this case I believe `Group` wouldn't be a dependency but a main contract.

So similar to mainchain folder I would suggest deploy the `Storage` and `OnlyOwnerAdminController` files here and then create a third deployment file `3_deploy_logic_contracts.js` Which then deploys the `Group`contract, this way deployment can upgrade the Group contract and then use the same storage.

Just double check with @craigwilliams84  on the above incase I got that wrong

Reply:
Yep, thats right.  If the `Group` contract is deployed in the same migration script as `Storage` then it won't be possible to deploy `Group` without deploying a new `Storage` contract as well.  They should be separated for sure.